generators/server/templates/build.gradle.ejs

```diff
@@ -474,8 +474,10 @@ dependencies {
     testCompile "com.github.differentway:couchbase-testcontainer"
     <%_ } _%>
     testCompile "org.hamcrest:hamcrest-library"
-    <%_ if (devDatabaseType !== 'h2Disk' && devDatabaseType !== 'h2Memory') { _%>
+    <%_ if (databaseType === 'sql') { _%>
```

Comment:
I don't think this is right, you have not taken into account sql database with no h2 and the same db in dev and prod.

Reply:
h2 is used during integration tests for sql database, isn't it ?

Reply:
In a SQL project, when you launch `mvn test`, it will use a h2 database for testing. That's why the dependency is here.

See the application.yml in test folder: https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/test/resources/config/application.yml.ejs#L61-L64